### ISSUE TITLE
Dataset problem 

### ISSUE THREAD

**Issue by ManojDir** [parthdasawant/Segmentation-of-Sclera#7]
Hi,
There are three daatset .Which dataset have you used for training?
SBVPI
1.Full dataset
2.segmentation subset
3.sclera subset

**Comment by parthdasawant** [comment]
HI @ManojDir,
Generally, you need actual and segmented mask images of the same images for training.
Can you share the screenshot of the dataset's folder so I can help you?

**Comment by ManojDir** [comment]
Hi @parthdasawant 
Which dataset do I need to download for training the model? can you provide that information?
![Screenshot_20231130-120045~2](https://github.com/parthdasawant/Segmentation-of-Sclera/assets/152391721/b1d6ce32-9190-4ebe-88ce-6ad8ca82b041)

**Comment by parthdasawant** [comment]
SBVPI

**Comment by ManojDir** [comment]
hi @parthdasawant 
I have these images from the SBVPI dataset. Which images do I have to add to the SS_train/images and SS_train/mask folders?
![Screenshot 2023-11-30 231201](https://github.com/parthdasawant/Segmentation-of-Sclera/assets/152391721/9b8b952d-4286-4df1-803e-67ae57fa3ef0)

**Comment by parthdasawant** [comment]
There should be two folders
1. Images: Contain the actual Images of the eye. e.g. 1L_l_2
2. Masks: Contains the segmented images(masks). e.g. 1L_l_2_sclera

Make sure that in both folders the images are renamed as 0.png, 1.png,...
and both folders have equal images in the proper order like 1L_l_2 should be images/0.png whereas 1L_l_2_sclera should be masks/0.png.

TIP: First of all try with the handful of images manually(since this is a 2-year-old code; Dependencies might be changed), if all works fine then try to write a script for that to automate the process.

**Comment by ManojDir** [comment]
Hi @parthdasawant 
But actual images are in JPG format , So how can we save them as PNG??

**Comment by parthdasawant** [comment]
You can use Pillow or OpenCV since you have to rename each image while making custom datasets so that time only you do this.

**Comment by patilbhavesh77** [comment]
I am a little bit confused. Can you please explain it in detail?

**Comment by ManojDir** [comment]
I am a little bit confused. Can you please explain it in detail?

**Comment by parthdasawant** [comment]
Sure @ManojDir Sorry for replying late,

You need two folders.
1. images: Real eye images
2. masks: Mask of those images

Now that you have the entire dataset set of eyes and multiple masks. You need to take only those images whose Sclera mask is available(I am assuming you are doing Segmentation of the Sclera) and put the actual image in the images folder and mask in the masks folder.

Make sure the images inside both folders are renamed like 0.png, 1.png.... etc.

For this repetitive work, you can use script using RegEx on the image name and separate the images and masks into respective folders. Then you can use those folders by uploading them to drive and passing the path of the folders to code.

If you have any further questions, please let me know.

**Comment by ManojDir** [comment]
Thank you, @parthdasawant for your invaluable guidence.

**Comment by ManojDir** [comment]
@parthdasawant 
![Screenshot 2024-02-13 171050](https://github.com/parthdasawant/Segmentation-of-Sclera/assets/152391721/0fbf520d-128d-4cfd-b31d-fa039671721f)
How can i solve this error?

**Comment by parthdasawant** [comment]
The shape of the logits and labels is mismatched at no. of channels; 1 means grayscale and 3 means RGB. Try reading/accessing the images in 3 channels (colorful) and not in grayscale mode. 

I am assuming the other code is unchanged. If you have made any changes, please share them with me so I can help you accordingly.

**Comment by ManojDir** [comment]
@parthdasawant 
[Model_Training.zip](https://github.com/parthdasawant/Segmentation-of-Sclera/files/14307409/Model_Training.zip)
This code i am using for model training.

**Comment by parthdasawant** [comment]
Did some tweaks. As I told you earlier it was due to the mismatching of the shapes.

<img width="1373" alt="Screenshot 2024-02-16 at 3 36 24 PM" src="https://github.com/parthdasawant/Segmentation-of-Sclera/assets/80618762/6bcc32bf-89a3-41df-8a30-5128018d413b">

You can find those changes by searching for `#changed` in the notebook.

Here is [Link](https://colab.research.google.com/drive/1eYLqzUaRFq5L2jD4eIMu7t0aYK1t2nfK?usp=sharing).

**Comment by ManojDir** [comment]
@parthdasawant 
The given link requires download permission to open the notebook.

**Comment by parthdasawant** [comment]
Please try again with request access.
